packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -202,10 +145,90 @@ module.exports = (
+                // of node ids which will then get re-parsed and the inferred schema
+                // recreated (if inferring schema gets too expensive, can also
+                // cache the schema until a query fails at which point recreate the
+                // schema).
+                //
+                // parse plugins take data from source nodes and extend it, never mutate
+                // it. Freeze all nodes once done so typegen plugins can't change it
+                // this lets us save off the DataTree at that point as well as create
+                // indexes.
+                //
+                // typegen plugins identify further types of data that should be lazily
+                // computed due to their expense, or are hard to infer graphql type
+                // (markdown ast), or are need user input in order to derive e.g.
+                // markdown headers or date fields.
+                //
+                // wrap all resolve functions to (a) auto-memoize and (b) cache to disk any
+                // resolve function that takes longer than ~10ms (do research on this
+                // e.g. how long reading/writing to cache takes), and (c) track which
+                // queries are based on which source nodes. Also if connection of what
+                // which are always rerun if their underlying nodes change..
+                //
+                // every node type in DataTree gets a schema type automatically.
+                // typegen plugins just modify the auto-generated types to add derived fields
+                // as well as computationally expensive fields.
+                if (process.env.NODE_ENV !== `production` || !fileNodes) {
+                  fileNodes = getNodes().filter(n => n.internal.type === `File`)
+                }
+                // Use Bluebird's Promise function "each" to run remark plugins serially.
+                Promise.each(pluginOptions.plugins, plugin => {
+                  const requiredPlugin = require(plugin.resolve)
+                  if (_.isFunction(requiredPlugin)) {
+                    return requiredPlugin(
+                      {
+                        markdownAST,
+                        markdownNode,
+                        getNode,
+                        files: fileNodes,
+                        pathPrefix,
+                        reporter,
+                        cache,
+                      },
+                      plugin.pluginOptions
+                    )
+                  } else {
+                    return Promise.resolve()
+                  }
+                }).then(() => {
+                  resolve(markdownAST)
+                })
               })
             })
-          })
+          }
+          const futureContent = processMarkdown(markdownNode.internal.content)
+          const futureExcerpt = markdownNode.excerpt
+            ? processMarkdown(markdownNode.excerpt)
```

Comment:
This will process excerpt markdown even if it won't be used - it would be better if this would be more lazy and only process markdown (both for excerpt and content) when someone is querying for it. It would need separate cache keys for content and excerpt